code/mob.dm

```diff
@@ -1665,7 +1666,11 @@
 				else
 					color_matrix_2_apply = mult_color_matrix(color_matrix_2_apply, src.color_matrices[cmatrix])
 			src.active_color_matrix = color_matrix_2_apply
-	src.client?.set_color(src.active_color_matrix)
+	if (src.client)
+		if (!src.respect_view_tint_settings)
+			src.client.set_color(src.active_color_matrix)
+		else
+			src.client.set_color(src.client.view_tint ? src.active_color_matrix : null)
```

Comment:
```suggestion
	src.client?.set_color(src.active_color_matrix, src.respect_view_tint_settings)
```

Reply:
Won't work, if respect_view_tint_settings is true, then it will set your color even if you have your view tint off